distribution/addon_distribution.py

```diff
@@ -0,0 +1,201 @@
+import os
+from enum import Enum
+from abc import abstractmethod
+import attr
+import logging
+import requests
+
+from distribution.file_handler import RemoteFileHandler
+
+
+class UrlType(Enum):
+    HTTP = "http"
+    GIT = "git"
+    OS = "os"
```

Comment:
I would call it `filesystem`

Reply:
or "local"? since you can't encode easily UNC paths in `file:` schema?

Reply:
I don't think there will be used schema for the type. In that case `addon_url` is just the path (which may be UNC) without `file:` at the start.